lib/internal/vm/module.js

```diff
@@ -382,6 +383,15 @@ class SyntheticModule extends Module {
       exportNames.some((e) => typeof e !== 'string')) {
       throw new ERR_INVALID_ARG_TYPE('exportNames', 'Array of strings',
                                      exportNames);
+    } else {
+      const syntheticExportNames = new Set();
+      for (const name of exportNames) {
+        if (syntheticExportNames.has(name)) {
+          throw new Error(`Export name ${name} has provided multiple times.`);
```

Comment:
this should throw an internal error with a code. maybe `ERR_INVALID_ARG_VALUE`?

Reply:
make sense